docs/extend/middleware.md

```diff
@@ -1 +1,108 @@
-#  Middleware
\ No newline at end of file
+#  Middleware
+
+Middleware is a nifty way to filter HTTP requests in Flarum. This can allow you to, for example, run a check against an external database checking a user when they register, either allowing them to continue the process or not. The possibilities are endless!
```

Comment:
I would suggest the following:

> Middleware is a nifty way to wrap the handling of HTTP requests in Flarum.

And then add an example that clarifies that you can do something to the request *before* handling as well as something to the response *after* handling.